database.c

```diff
@@ -1322,3 +1335,27 @@ database_free_changed_values(profileval_t *values)
 {
   profileval_free_vector(values);
 }
+
+void
+migrate_configs(const char *legacy, const char *current)
+{
+  int status;
+  pid_t id = fork();
+  if (id < 0)
+  {
+    log_err("Forking failed with error %d\n", id);
+  }
+  else if (id == 0)
+  {
+    execlp("mv", "mv", "-f", legacy, current, NULL);
```

Comment:
Should be cleaner moving files in-process instead of forking and executing command line tools. Glib surely has nice options for that.

Besides it's possibly a timing problem if the moving happens externally.